### PR TITLE
chore: build on linux and windows on GH actions

### PR DESCRIPTION
**Pull request by rumpl**
Things I did:

* added windows runner on github actions
* made them required before merging
* added the badge in the readme file
* removed codecov from dependencies, use codecov action for this
* removed travis build (travis will be only used for deploying)
* removed appveyor

Things left to do:
* move the deployment over to github actions too

### CODE REVIEW

**Comment by LinusU** [comment]
Neat! 👍 

Should probably change the `name: linux ci` also

**Comment by rumpl** [comment]
@LinusU yeah will rename, ideally we wouldn't need tavis and appveyor any more, will do some tests.

**Comment by rumpl** [comment]
I already tried gituhb actions on windows but they were flaky at the time, hopefully it's better now

**Comment by rumpl** [comment]
Breaks because of symlinks :'(

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -26,7 +35,12 @@ jobs:
           npm run lint
           npm run prettier-check
           npm run test-coverage
-          npm run test-dependent
```
Comment (LinusU) [inline .github/workflows/nodejs.yml:29]: Doesn't seem like this is run from anywhere anymore, should we also remove the package.json script & dependency?

Comment (rumpl) [reply .github/workflows/nodejs.yml:29]: Done 👍

**Pull request merged**